ecms_base/modules/custom/ecms_api_publisher/src/Plugin/QueueWorker/EcmsApiSyndicateQueueWorker.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\ecms_api_publisher\Plugin\QueueWorker;
+
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\Core\Queue\QueueWorkerBase;
+use Drupal\Core\Queue\RequeueException;
+use Drupal\Core\StringTranslation\StringTranslationTrait;
+use Drupal\ecms_api_publisher\EcmsApiPublisher;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Post syndicated content on cron.
+ *
+ * @QueueWorker(
+ *   id = "ecms_api_publisher_queue",
+ *   title = @Translation("Ecms API Cron Publisher"),
+ *   cron = {"time" = 5}
+ * )
+ */
+class EcmsApiSyndicateQueueWorker extends QueueWorkerBase implements ContainerFactoryPluginInterface {
+
+  use StringTranslationTrait;
+
+  /**
+   * The ecms_api_publisher.publisher service.
+   *
+   * @var \Drupal\ecms_api_publisher\EcmsApiPublisher
+   */
+  private $ecmsApiPublisher;
+
+  /**
+   * {@inheritDoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('ecms_api_publisher.publisher')
+    );
+  }
+
+  /**
+   * EcmsApiSyndicateQueueWorker constructor.
+   *
+   * @param array $configuration
+   *   A configuration array containing information about the plugin instance.
+   * @param string $plugin_id
+   *   The plugin ID for the plugin instance.
+   * @param mixed $plugin_definition
+   *   The plugin implementation definition.
+   * @param \Drupal\ecms_api_publisher\EcmsApiPublisher $ecmsApiPublisher
+   *   The ecms_api_publisher.publisher service.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, EcmsApiPublisher $ecmsApiPublisher) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+
+    $this->ecmsApiPublisher = $ecmsApiPublisher;
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  public function processItem($data) {
```

Comment:
Argument type and return type

Reply:
I added the return type hint here. I can't typehint the variable as it breaks the `QueuWorkerInterface` syntax. I did update the comment to reflect what it should be instead.

Reply:
👍